src/pages/index.js

```diff
@@ -40,15 +80,84 @@ function createCardElement(data) {
   cardImage.alt = data.name;
   cardTitle.textContent = data.name;
 
-  cardLikedBtn.addEventListener("click", () => cardLikedBtn.classList.toggle("card__like-btn--liked"));
+  if (data.isLiked) {
+    cardLikedBtn.classList.add("card__like-btn--liked");
+  }
+
+  cardLikedBtn.addEventListener("click", () => {
+    const isCurrentlyLiked = cardLikedBtn.classList.contains("card__like-btn--liked");
+
+    api.toggleLike(data._id, !isCurrentlyLiked)
+      .then(updatedCard => {
+        cardLikedBtn.classList.toggle("card__like-btn--liked", updatedCard.isLiked);
+      })
+      .catch(err => console.error("Error toggling like:", err));
+  });
+
   cardImage.addEventListener("click", () => openImageModal(data.link, data.name));
-  deleteBtn.addEventListener("click", (e) => { e.stopPropagation(); cardElement.remove(); });
+
+  deleteBtn.addEventListener("click", (e) => {
+    e.stopPropagation();
+    selectedCard = cardElement;
+    selectedCardId = data._id;
+    openPopup(deleteModal);
+  });
 
   return cardElement;
 }
 
+deleteConfirmBtn.addEventListener("click", (event) => {
+  event.preventDefault();
+  if (selectedCardId) {
+    const originalText = deleteConfirmBtn.textContent;
+    deleteConfirmBtn.textContent = "Deleting...";
+
+    api.deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        selectedCard = null;
+        selectedCardId = null;
+        closePopup(deleteModal);
+      })
+      .catch(err => console.error("Error deleting card:", err))
+      .finally(() => {
+        deleteConfirmBtn.textContent = originalText;
+      });
+  }
+});
+
+deleteCancelBtn.addEventListener("click", () => {
+  closePopup(deleteModal);
+  selectedCard = null;
+  selectedCardId = null;
+});
+
+avatarEditBtn.addEventListener("click", () => {
+  enableValidation(Settings);
```

Comment:
Please, delete it from the `avatar` modal opening

`enableValidation` should be called only 1 time to enable validation. You should not do that when you open/close a popup because you add more and more the same listeners to the same forms. This may cause a `memory leak`

 (you have it in line 249 already)